src/txn/txn_rollback_to_stable.c

```diff
@@ -1694,22 +1695,26 @@ __rollback_to_stable_btree_apply(
           __wt_timestamp_to_string(rollback_timestamp, ts_string[1]), rollback_txnid);
 
     /*
-     * Truncate history store entries for the non-timestamped table.
-     * Exceptions:
+     * Truncate history store entries for the non-timestamped table or if the table doesn't exist.
+     *
+     * Exceptions for non-timestamped table:
      * 1. Modified tree - Scenarios where the tree is never checkpointed lead to zero
      * durable timestamp even they are timestamped tables. Until we have a special
      * indication of letting to know the table type other than checking checkpointed durable
      * timestamp to WT_TS_NONE, we need this exception.
      * 2. In-memory database - In this scenario, there is no history store to truncate.
      */
-    if ((!dhandle_allocated || !S2BT(session)->modified) && max_durable_ts == WT_TS_NONE &&
-      !F_ISSET(S2C(session), WT_CONN_IN_MEMORY)) {
+    exist = __wt_metadata_search(session, uri, &metadata_conf) != WT_NOTFOUND;
```

Comment:
@kommiharibabu  and I both talked about this, and realised that we needed to go through the list of entries that do not exist in the metadata, so that we can clean the history store content for those entries.